src/player_display.cpp

```diff
@@ -96,6 +96,55 @@ static std::vector<std::pair<bodypart_id, bool>> list_and_combine_bps( const Cha
     return bps;
 }
 
+static std::pair<size_t, size_t> subindex_around_cursor(
+    const size_t list_lenght,
+    const size_t available_space,
+    const size_t cursor_pos,
+    const bool focused
+)
+/**
+ * Return indexes [start, end) that should be displayed from list long `list_lenght`,
+ * given that cursor is at position `cursor_pos` and we have `available_space` spaces.
+ *
+ * Example:
+ * list_lenght = 6, available_space = 3, cursor_pos = 2, focused = true;
+ * so choose 3 from indexes [0, 1, 2, 3, 4, 5]
+ * return {1, 4}
+ */
+{
+    if( !focused || list_lenght <= available_space ) {
+        return std::make_pair( 0, std::min( available_space, list_lenght ) );
+    }
+    size_t slice_start = std::max( 0, int( cursor_pos ) - int( available_space ) / 2 );
+    slice_start = std::min( slice_start, list_lenght - available_space );
+    size_t slice_end = slice_start + available_space;
+    return std::make_pair( slice_start, slice_end );
+}
+
+static void draw_scrollbar(
+    const size_t list_lenght,
+    const size_t height,
+    const size_t width,
+    const size_t height_offset,
+    const size_t range_first,
+    const catacurses::window &win
+)
+/**
+ * Draw scrollbar if scrollable.
+ */
+{
+    if( list_lenght > height ) {
+        scrollbar()
+        .offset_x( width )
+        .offset_y( height_offset )
+        .content_size( list_lenght )
+        .viewport_pos( range_first )
+        .viewport_size( height )
+        .scroll_to_last( false )
+        .apply( win );
+    }
+}
```

Comment:
The intention of using the `scrollbar` class is that it's clear from sight what the arguments are for. If you really want less code duplication, there's `draw_scrollbar` from `output.h`, but that function is sort of deprecated.

Reply:
To be honest, I named my function the same, so I should probably change at least that.
I was hoping that eventually, only the body of for loop would survive in these functions returning iterator, so this function would be called only once as my other function. Then they both would be expanded and removed...

Reply:
This seems to be the last thing. I would like to keep it as-is. But I didn't really resolve the issue mentioned, so I will not mark this as resolved.

Reply:
Ok, I replaced it with the legacy function. Funnily enough, the `if` in  `if (do_draw_scrollbar) {draw_scrollbar(...);}` was useless, since it's handled in `class scrollbar::apply`.